File: src/my_py_pkg/my_py_pkg/my_first_node.py
#/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("my_first_node")
        self.count_ = 0
        self.create_timer(1, self.timer_callback)
        self.get_logger().info("My First Node Started!")

    def timer_callback(self):
        self.get_logger().info("Hello " + str(self.count_))
        self.count_ += 1


def main(args=None):
    rclpy.init()
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include <rclcpp/rclcpp.hpp>

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node{"my_first_node"}
    {
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), [this](){
            this->timer_callback();
        });
        RCLCPP_INFO(this->get_logger(), "My First Node Started!");
    }

    void timer_callback() {
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_++);
    }

private:
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}